examples/abitest/client/cpp/httptest.cc

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright 2020 The Project Oak Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "examples/abitest/client/cpp/httptest.h"
+
+#include "glog/logging.h"
+#include "httplib.h"
+#include "oak_abi/proto/label.pb.h"
+
+const char* CA_CERT_PATH = "../../../../../../../../../examples/certs/local/ca.pem";
+
+// Simple manual test case registry.
+const std::map<std::string, HttpTestFn> http_tests = {
+    {"HttpsWithJsonLabelOk", test_https_with_json_label_ok},
+    {"HttpsWithProtobufLabelOk", test_https_with_protobuf_label_ok},
+    {"HttpsWithoutLabelErrBadRequest", test_https_without_label_err_bad_request},
+    {"UnsecureHttpErr", test_unsecure_http_err},
+};
+
+bool test_https_with_json_label_ok() {
+  httplib::SSLClient cli("localhost", 8383);
+  cli.set_ca_cert_path(CA_CERT_PATH);
+  cli.enable_server_certificate_verification(true);
+  httplib::Headers headers = {{"oak-label", "{\"confidentialityTags\":[],\"integrityTags\":[]}"}};
```

Comment:
Could / should we move these tests to Rust at some point?

Reply:
Yeah I was just thinking, I don't think it's urgent, but AFAICT the reason why these tests are C++ is mostly historical, so maybe it would make sense to port them to Rust at some point, for consistency with the rest of the codebase.